RBA Migration

ljstella opened the pull request:
### Details

Content migration accompanying changes to contentctl: https://github.com/splunk/contentctl/pull/263



### Checklist

- [ ] Validate name matches `<platform>_<mitre att&ck technique>_<short description>` nomenclature
- [ ] [CI/CD](https://github.com/splunk/security_content/actions) jobs passed ✔️ 
- [ ] Validated SPL logic.
- [ ] Validated tags, description, and how to implement.
- [ ] Verified references match analytic.
- [ ] Confirm updates to lookups are handled properly.

### Notes For Submitters and Reviewers

- If you're submitting a PR from a fork, ensuring the box to allow updates from maintainers is checked will help speed up the process of getting it merged.
- Checking the output of the `build` CI job when it fails will likely show an error about what is failing. You may have a very descriptive error of the specific field(s) in the specific file(s) that is causing an issue. In some cases, its also possible there is an issue with the YAML. Many of these can be caught with the pre-commit hooks if you set them up. These errors will be less descriptive as to what exactly is wrong, but will give you a column and row position in a specific file where the YAML processing breaks. If you're having trouble with this, feel free to add a comment to your PR tagging one of the maintainers and we'll be happy to help troubleshoot it.
- Updates to existing lookup files can be tricky, because of how Splunk handles application updates and the differences between existing lookup files being updated vs new lookups. You can read more [here](https://docs.splunk.com/Documentation/SplunkCloud/8.2.2203/Admin/PrivateApps#Manage_lookups_in_Splunk_Cloud_Platform) but the short version is that any changes to lookup files need to bump the datestamp in the lookup CSV filename, and the reference to it in the YAML needs to be updated.

ljstella committed:
Application detections

ljstella committed:
Merge branch 'develop' into rba_migration

patel-bhavin committed:
Branch was auto-updated.

patel-bhavin committed:
Branch was auto-updated.

patel-bhavin committed:
Branch was auto-updated.

ljstella committed:
endpoint first pass

ljstella committed:
cloud detections initial translation

ljstella committed:
Merge branch 'develop' into rba_migration

ljstella committed:
Deprecated Detections initial translation

ljstella committed:
network detections initial translation

ljstella committed:
web detections initial translation

ljstella committed:
endpoint detection score fix

ljstella committed:
cloud detection score fix

ljstella committed:
deprecated detection score fix

ljstella committed:
network detections score fix

ljstella committed:
web detections score fix

ljstella committed:
application detections score fix

ljstella committed:
Application detection score field rename

ljstella committed:
endpoint detection score field rename

ljstella committed:
web detection score field rename

ljstella committed:
cloud detection score field rename

ljstella committed:
deprecated detection score field rename

ljstella committed:
network detection score field rename

ljstella committed:
missing types

ljstella committed:
endpoint: lowercase rba types

ljstella committed:
web: lowercase rba types

ljstella committed:
cloud: lowercase rba types

ljstella committed:
network: lowercase rba types

ljstella committed:
deprecated: lowercase rba types

ljstella committed:
application: lowercase rba types

ljstella committed:
web: ip address typefix

ljstella committed:
endpoint: ip address typefix

ljstella committed:
deprecated: ip address typefix

ljstella committed:
cloud: ip address typefix

ljstella committed:
application: ip address typefix

ljstella committed:
network: ip address typefix

ljstella committed:
application: more typefixes

ljstella committed:
endpoint: more typefixes

ljstella committed:
cloud: more typefixes

ljstella committed:
deprecated: more typefixes

ljstella committed:
network: more typefixes

ljstella committed:
web: more typefixes

ljstella committed:
artifact of cleanup

ljstella committed:
application: more typefixes

ljstella committed:
cloud: more typefixes

ljstella committed:
deprecated: more typefixes

ljstella committed:
endpoint: more typefixes

ljstella committed:
network: more typefixes

ljstella committed:
web: more typefixes

ljstella committed:
web: remove rba from hunting

ljstella committed:
endpoint: remove rba from hunting

ljstella committed:
application: remove rba from hunting

ljstella committed:
cloud: remove rba from hunting

ljstella committed:
deprecated: remove rba from hunting

ljstella committed:
network: remove rba from hunting

ljstella committed:
Cleanup rba

ljstella committed:
cloud: more cleanup

ljstella committed:
endpoint: more cleanup

ljstella committed:
web: more cleanup

ljstella committed:
application: remove rba config from correlations

ljstella committed:
cloud: remove rba config from correlations

ljstella committed:
deprecated: remove rba config from correlations

ljstella committed:
endpoint: remove rba config from correlations

ljstella committed:
web: remove rba config from correlations

ljstella commented:
Status update:

I have (temporarily) changed the GitHub Actions config in my PR (https://github.com/splunk/contentctl/pull/263) to run the ESCU smoketest against this branch. 

Here are the latest results: https://github.com/splunk/contentctl/actions/runs/11860779327/job/33057262480?pr=263

As of this point, there are now under 100 validation errors against translated content. 

The vast majority of these will need to be handled individually. Up until this point, commits have generally been done by subdirectory of `detections/` with all of the changes for those in each one to assist in review. If you're reviewing this change at a later point, this is a great time to go refill your coffee ☕  and grab a snack 🍿 . I will attempt to group similar fixes and/or subdirectories together from this point forward, but they may be a bit more spread out.

ljstella committed:
application: threat object type cleanup

ljstella committed:
cloud: threat object type cleanup

ljstella committed:
deprecated: threat object type cleanup

ljstella committed:
endpoint: threat object type cleanup

ljstella committed:
network: threat object type cleanup

ljstella committed:
web: threat object type cleanup

ljstella committed:
stragglers cleanup

ljstella commented:
Update: as of now, **all** translated content is building successfully: https://github.com/splunk/contentctl/actions/runs/11863423975?pr=263

Still have to do some cleanup around messages and some risk objects, but they're all technically passing and producing a package "compatible" with the previous releases. I say compatible, as opposed to identical, because as this work was done, some changes have been made where things were threat objects but probably should have been risk objects, vice versa, and some where they just didn't make sense as either and were removed. Some risk messages were updated along the way, although there will be more of that upcoming.

cmcginley-splunk commented:
This isn't a fully formed thought, but considering the recent discussions we've had around the notables and risk analysis events in ES 8.0, do we think it makes sense to lump notables under the terminology `risk_objects`? Does it make sense to explicitly distinguish risk analysis objects from notable objects in the YAML? I'm not convinced it does, just wanna raise a convo around it

ljstella commented:
@cmcginley-splunk 
> do we think it makes sense to lump notables under the terminology risk_objects?

Could you expand on this a bit? I don't think we have anything that does this today, but generally speaking (at least, pre-ES8) notables were **extremely** simple, especially in comparison to risk events, in form of schema, required bits and bobs, and most importantly, a simple 1:1 relationship between search results and journaled events. 

Some of the drivers behind these changes like aligning the terminology in the YAML with that in the product, and a more simplistic representation of the objects in config with fewer levels of abstraction seem at odds with the idea of putting notable-related things into an `rba` object.

Do I think that at a future point, we should consider a similar effort with a `notable` object that configures all of the `action.notable.param` fields based on the detection's deployment? Perhaps, although we can sort of already do that with the in-line deployments.

ljstella committed:
Merge branch 'develop' into rba_migration

ljstella committed:
Updates with develop

ljstella committed:
application: cleanup tbd messages

ljstella committed:
Cloud: cleanup tbd messages

ljstella committed:
Endpoint: cleanup TBD messages

ljstella committed:
network: cleanup tbd messages

ljstella committed:
web: cleanup TBD messages

ljstella committed:
deprecated: cleanup TBD messages

ljstella committed:
cloud: cleanup of risk object type other

ljstella committed:
endpoint: cleanup of risk object type other

ljstella committed:
cloud: missed one

patel-bhavin committed:
Branch was auto-updated.

patel-bhavin committed:
Branch was auto-updated.

ljstella committed:
Merge branch 'develop' into rba_migration

ljstella committed:
Merge branch 'develop' into rba_migration

ljstella committed:
rba conversion

patel-bhavin committed:
Branch was auto-updated.

ljstella committed:
small grammar fix

ljstella committed:
Merge branch 'develop' into rba_migration

Commit:
Improved ASL AWS detections

Commit:
bug fixes

Commit:
bug fixes

Commit:
new asl aws detections

Commit:
bug fix

Commit:
bug fix

Commit:
bug fix

Commit:
bug fix

Commit:
bug fix

Commit:
bug fix improvements

Commit:
updates

Commit:
bug fix

Commit:
bug fix

Commit:
bug fix

Commit:
new detection

patel-bhavin committed:
Branch was auto-updated.

ljstella committed:
Rename to A&I compatible user field

ljstella committed:
Merge branch 'develop' into rba_migration

ljstella committed:
Merge branch 'develop' into rba_migration

ljstella committed:
Merge branch 'develop' into rba_migration

Commit:
new detection

Commit:
bug fix

Commit:
new detection

ljstella committed:
Merge branch 'develop' into rba_migration

ljstella committed:
Merge branch 'develop' into rba_migration

pyth0n1c committed:
first batch of updates to
lookup ymls

pyth0n1c committed:
Second round of lookup file cleanup

pyth0n1c committed:
another round of lookup file updates

pyth0n1c committed:
More cleanup of lookups

pyth0n1c committed:
more cleanup of lookups to follow new format.
this involved renaming some CSVs to remove the
dates from them or force their naming schema to
match the corresponding YML name for CSV
detections

pyth0n1c committed:
finish lookup cleanup

pyth0n1c committed:
removes types from stories

pyth0n1c committed:
remove extra fields from baselines and investigations

patel-bhavin committed:
Branch was auto-updated.

pyth0n1c committed:
remove collection for all kvstores.  this will just come from the name. also ensure the each kvstore begins with _key.

pyth0n1c committed:
Merge branch 'rba_migration' into strict_yml_from_rba

patel-bhavin committed:
Branch was auto-updated.

pyth0n1c committed:
remove update_timestamps, confidence, impact,
related_fields, and risk_score from detections

pyth0n1c committed:
Finish removing extra fields, or renaming
misnamed fields, in endpoint detections

pyth0n1c committed:
progress on input/output lookup
cleanup after new validations
introduced

dluxtron committed:
Uploading new detections

dluxtron committed:
Uploading new detections

dluxtron committed:
Adding intune detections & updating dataset links

dluxtron committed:
fixing syntax & updating macro

ljstella committed:
Stragglers migration

ljstella committed:
Merge branch 'develop' into rba_migration

patel-bhavin committed:
fixing up yamls for testing

patel-bhavin committed:
updating risk

patel-bhavin committed:
updating macro

patel-bhavin committed:
udpating SPL

dluxtron committed:
reverting macro change

Commit:
new detection

Commit:
bug fix

Commit:
new detection

patel-bhavin committed:
Branch was auto-updated.

Commit:
new detections

Commit:
change

Commit:
improvements

patel-bhavin committed:
Branch was auto-updated.

pyth0n1c committed:
Merge branch 'rba_migration' into strict_yml_from_rba

ljstella committed:
More stragglers

patel-bhavin committed:
Branch was auto-updated.

pyth0n1c committed:
remove confidence impact message

pyth0n1c committed:
Merge branch 'rba_migration' into strict_yml_from_rba

pyth0n1c committed:
Merge branch 'strict_yml_from_rba' of https://github.com/splunk/security_content into strict_yml_from_rba

patel-bhavin committed:
Branch was auto-updated.

patel-bhavin committed:
Branch was auto-updated.

patel-bhavin committed:
Branch was auto-updated.

patel-bhavin committed:
Branch was auto-updated.

patel-bhavin committed:
minor udpates to yaml

ljstella committed:
New detection conversion

patel-bhavin committed:
Branch was auto-updated.

pyth0n1c committed:
Merge branch 'rba_migration' into strict_yml_from_rba

patel-bhavin committed:
Update o365_service_principal_privilege_escalation.yml

patel-bhavin committed:
updating data sources

Commit:
Add ASL AWS CloudTrail data source

pyth0n1c committed:
remove extra fields

pyth0n1c committed:
Merge branch 'strict_yml_from_rba' of https://github.com/splunk/security_content into strict_yml_from_rba

ljstella committed:
Merge branch 'develop' into rba_migration

ljstella committed:
Merge branch 'rba_migration' into strict_yml_from_rba

ljstella committed:
More migrations: AWS, Azure, O365

patel-bhavin committed:
Branch was auto-updated.

ljstella committed:
Removal of fields from new detections

ljstella committed:
Merge branch 'develop' into rba_migration

pyth0n1c committed:
Fix more errors with missing lookups, baselines, and detections

ljstella committed:
Merge pull request #3269 from splunk/strict_yml_from_rba

Strict yml from rba

pyth0n1c committed:
fix baseline which referenced lookups
that do not exist

patel-bhavin commented:
@ljstella : Shall we update the github workflows build, inpect and test to install the alpha version of contentctl?

ljstella commented:
> @ljstella : Shall we update the github workflows build, inpect and test to install the alpha version of contentctl?

Once there is an alpha release of contentctl 5.0.0, these are the spots ~it needs to change~ (actually, maybe doesn't need to change with that): 
 
Here: 
https://github.com/splunk/security_content/blob/863eb07dc93edfa87c594f5c0fbbedf670e032b2/.github/workflows/build.yml#L22

Here:
https://github.com/splunk/security_content/blob/863eb07dc93edfa87c594f5c0fbbedf670e032b2/.github/workflows/unit-testing.yml#L26

Here: https://github.com/splunk/security_content/blob/863eb07dc93edfa87c594f5c0fbbedf670e032b2/.github/workflows/appinspect.yml#L21

Here: 
https://github.com/splunk/security_content/blob/863eb07dc93edfa87c594f5c0fbbedf670e032b2/.github/workflows/datasource-dependabot.yml#L27

patel-bhavin committed:
empty commit

patel-bhavin committed:
specify ctl version

patel-bhavin committed:
bump to alpha2

patel-bhavin commented:
App inspect changes works locally ! LGTM
